src/ast/semantic_analyser.cpp

```diff
@@ -645,6 +645,18 @@ void SemanticAnalyser::visit(Call &call)
     buf << "BPF_FUNC_send_signal not available for your kernel version";
 #endif
   }
+  else if (call.func == "strncmp") {
+    if (check_nargs(call, 3)) {
+      check_arg(call, Type::string, 0);
+      check_arg(call, Type::string, 1);
+      check_arg(call, Type::integer, 2, true);
+
+      Integer &size = static_cast<Integer&>(*call.vargs->at(2));
```

Comment:
At this point it's not guaranteed that the third argument is an integer literal, so this could lead to some dodgy memory accesses. This section needs to only run if `check_arg(call, Type::integer, 2, true)` returns true